src/Core/src/Platform/Windows/ContentPanel.cs

```diff
@@ -97,18 +97,32 @@ public void UpdateBackground(Paint? background)
 			_borderPath.UpdateBackground(background);
 		}
 
+		[Obsolete("Use Microsoft.Maui.Platform.UpdateBorderStroke instead")]
 		public void UpdateBorderShape(IShape borderShape)
 		{
-			_borderShape = borderShape;
+			if (borderShape is null || _borderPath is null)
+				return;
+
+			_borderPath.UpdateBorderShape(borderShape, ActualWidth, ActualHeight);
```

Comment:
I would refactor this to a private method so it can be called from here and `UpdateBorderStroke`

Reply:
Nice suggestion, created.